db-tests/src/db_tests.rs

```diff
@@ -174,7 +174,7 @@ async fn get_bsos_limit_offset() -> Result<()> {
         .compat()
         .await?;
     assert_eq!(bsos.items.len(), 5 as usize);
-    assert_eq!(bsos.offset, Some(5));
+    assert_eq!(bsos.offset, Some("5".to_string()));
```

Comment:
For testing, I would think some of these cases would already hit the new offset code? If not, we could add some and test for encoded offset values when running under spanner.

(The Db could have some elaborate `supports_encoded_offset() -> bool` method, but I'd be fine with a quick check like the one recently added here: https://github.com/mozilla-services/syncstorage-rs/blob/70bfcbb0d6c649d285b106edbf6bb3b06615fc19/src/db/mysql/test.rs#L57 )

Reply:
Additionally we would need some spanner specific tests for testing decoding of offsets

Reply:
Ok, in the process of fixing the last test I did add code which tests the results of the spanner offset optimization.